### PR TITLE
Add allowExpressions option

### PR DESCRIPTION
**Pull request by matthewloring**

### CODE REVIEW

File: test/test-v8debugapi.js [hunk 1]
```diff
@@ -915,6 +916,30 @@ describe('v8debugapi', function() {
         });
     });
 
+    it('should capture without values when `allowExpressions=false`', function(done) {
+      // clone a clean breakpointInFoo
+      var bp = {id: breakpointInFoo.id, location: breakpointInFoo.location};
+      config.allowExpressions = false;
```
Comment (ofrobots) [inline test/test-v8debugapi.js]: better: `var config = extend(config, { allowExpressions: false })`

File: test/test-debuglet.js [hunk 2]
```diff
@@ -526,6 +527,88 @@ describe('Debuglet', function() {
       debuglet.start();
     });
 
+    it('should reject breakpoints with conditions when allowExpressions=false',
```
Comment (ofrobots) [inline test/test-debuglet.js]: nit: It would be better+simpler to add this test to `test-v8debug.js`.

File: test/test-debuglet.js [hunk 3]
```diff
@@ -526,6 +527,88 @@ describe('Debuglet', function() {
...
+          debuglet.config_.allowExpressions = true;
+          scope.done();
+          done();
+        }, 1000);
+      });
+
+      debuglet.start();
+    });
+
+    it('should reject breakpoints with expressions when allowExpressions=false',
```
Comment (ofrobots) [inline test/test-debuglet.js]: ditto.

File: src/agent/state.js [hunk 4]
```diff
@@ -290,7 +296,16 @@ StateResolver.prototype.isPathInNodeModulesDirectory_ = function(path) {
...
   var args = [];
   var locals = [];
-  if (!underFrameCap) {
+  if (!this.config_.allowExpressions) {
+    args.push({
+      name: 'arguments_not_available',
+      varTableIndex: EXPRESSIONS_DISALLOWED_INDEX
+    });
+    locals.push({
+      name: 'locals_not_available',
```
Comment (ofrobots) [inline src/agent/state.js]: Remind me again why looking at local values can have side effects? I do agree that we shouldn't look at object properties.

File: test/test-v8debugapi.js (hunk 1 above)
Comment (matthewloring) [inline test/test-v8debugapi.js]: That creates a new `config` object. This works by updating the values on the global config which is held by reference in the `api`. If we use `extend` we will need comparable lines replacing/restoring the config object held by the api.

File: test/test-debuglet.js (hunk 2 above)
Comment (matthewloring) [inline test/test-debuglet.js]: I can't find a `test-v8debug.js`, do you mean `test-v8debugapi.js`? The issue is that the logic being testing is in `debuglet.js` so I'm not sure it is testable from `test-v8debugapi.js`.

File: test/test-debuglet.js (hunk 3 above)
Comment (matthewloring) [inline test/test-debuglet.js]: same as above.

File: src/agent/state.js (hunk 4 above)
Comment (matthewloring) [inline src/agent/state.js]: I remember talking through it when we talked about the behavior of the new option being added but I don't recall the reason. We already avoid interceptors/getters on objects by default so object properties should be ok if locals are in general?

Comment (matthewloring) [inline src/agent/state.js]: After digging back through some of the frame mirror resolution logic, I think locals/args are fine. I've added a comment justifying this.

File: src/agent/config.js [hunk 5]
```diff
@@ -37,6 +37,15 @@ module.exports = {
    */
   description: null,
 
+  /**
+   * @property {boolean} Whether or not it is safe to evaluate expressions.
```
Comment (ofrobots) [inline src/agent/config.js]: nit: s/safe/permitted

File: src/agent/state.js [hunk 6]
```diff
@@ -290,6 +290,8 @@ StateResolver.prototype.isPathInNodeModulesDirectory_ = function(path) {
 StateResolver.prototype.resolveFrame_ = function(frame, underFrameCap) {
   var args = [];
   var locals = [];
+  // Locals and arguments are safe to collect even when `config.allowExpressions=false`
+  // since we properly avoid inspecting interceptors and getters by default.
```
Comment (ofrobots) [inline src/agent/state.js]: Are there tests with a getter, setter, interceptors that will keep us sane?

Comment (matthewloring) [inline src/agent/state.js]: Yup: https://github.com/GoogleCloudPlatform/cloud-debug-nodejs/blob/48a9952ec004e18acb9e5ca876a411493b742129/test/test-v8debugapi.js#L657

File: src/agent/config.js (hunk 5 above)
Comment (matthewloring) [inline src/agent/config.js]: done

**Pull request merged**